vulcanai2/models/basenetwork.py

```diff
@@ -465,46 +454,75 @@ def forward_pass(self, input_data, convert_to_class=False):
         else:
             return output
 
-    # TODO: this is copy pasted - edit as appropriate
-    def save_model(self, save_path='models'):
+    def save_model(self, save_path=None):
         """
-        Will save the model parameters to a npz file.
-        Args:
-            save_path: the location where you want to save the params
+        Save the model (and its' input networks)
+        :param save_path: The save directory (not a file)
+        :return: save path, for recursive purposes
         """
-        if self.input_network is not None:
-            if not hasattr(self.input_network['network'], 'save_name'):
-                self.input_network['network'].save_model()
 
-        if not os.path.exists(save_path):
-            print('Path not found, creating {}'.format(save_path))
-            os.makedirs(save_path)
-        file_path = os.path.join(save_path, "{}{}".format(self.timestamp,
-                                                          self.name))
-        save_name = '{}.network'.format(file_path)
-        print('Saving model as: {}'.format(save_name))
+        print("in save", self.epoch)
+
+        if not save_path:
+            save_path = "{date:%Y-%m-%d_%H:%M:%S}/".format(name=self.name, date=datetime.now())
+            logger.info("No save path provideded, saving to {}".format(save_path))
+
+        if not save_path.endswith("/"): #TODO: does this break windows?? no idea.
+            save_path = save_path + "/"
+
+        module_save_path = save_path + "{name}/".format(name=self.name)
+
+        os.makedirs(module_save_path) #let this throw an error if it already exists
+
+        #recursive recursive recursive
+        if self._input_network is not None:
+            self._input_network.save_model(save_path)
 
-        with open(save_name, 'wb') as f:
-            pickle.dump(self, f, protocol=pickle.HIGHEST_PROTOCOL)
+        self.save_path = save_path #TODO: I don't think this is necessary
 
-        self.save_metadata(file_path)
+        # to improve: # object.__getstate__() https://docs.python.org/3/library/pickle.html#example
+        model_file_path = module_save_path + "model.pkl"
+        state_dict_file_path = module_save_path + "state_dict.pkl"
+        pickle.dump(self, open(model_file_path, "wb"), 2)
+        pickle.dump(self.state_dict, open(state_dict_file_path, "wb"), 2) #TODO: pretty sure this isn't necessary
 
+    #TODO: update the state dict and push to the appropriate device.
+    #https://pytorch.org/tutorials/beginner/saving_loading_models.html#what-is-a-state-dict
     @classmethod
-    def load_model(cls, load_path):
+    def load_ensemble(cls, load_path):
```

Comment:
let's generalize it to be just load_model() so it should be able to load individual and multi-component networks